### PR TITLE
Support certificate expiry on bot join

### LINKED ISSUE #16691: Bots joined via IAM tokens ignore user TTL
Expected behavior:

Bots joined via AWS IAM tokens should honor requested certificate TTLs.

Current behavior:

Certificates always have a 1 hour TTL.

This is caused by behavior differences in IAM-joined bots. With regular token joining, we issue the first set of renewable certs (generation 0) through a special joining endpoint [that does not accept a user-defined TTL](https://github.com/gravitational/teleport/blob/45c065aceed664f747532c7ca9463225b86079b9/lib/auth/join.go#L138). These certs are immediately used to issue a fresh set of certificates _with_ a user-defined TTL (generation 1), and the original (generation 0) certs are thrown away. 

However, with IAM joining, both initial join and renewals just use the initial (generation 0) join process. No second set of certificates is ever issued, so only the hard-coded TTL is ever used. Unfortunately this means the user's requested TTL is never honored.

Bug details:
- Teleport version: v10.2
- Recreation steps
- Debug logs

### PR DESCRIPTION
**Pull request by strideynet**
Closes https://github.com/gravitational/teleport/issues/16691

Allows an expiry time for generated certificates to be requested on Join. This fixes certificate expiries being incorrect for GitHub and IAM joining.

### CODE REVIEW

File: lib/auth/join_test.go [hunk 1]
```diff
@@ -367,3 +367,81 @@ func TestRegister_Bot(t *testing.T) {
...
+			requestExpires: &validExpires,
+			expectTTL:      time.Hour * 6,
+		},
+		{
+			name:           "value exceeding limit specified",
+			requestExpires: &tooGreatExpires,
+			// MaxSessionTTL set in createBotRole is 12 hours, so this cap will
+			// apply instead of the defaults.MaxRenewableCertTTL specified
+			// in generateInitialBotCerts.
+			expectTTL: 12 * time.Hour,
```
Comment (strideynet) [inline lib/auth/join_test.go:408]: This is a little unexpected. Looking at the implementation of `generateInitialBotCerts` you'd expect this to be 24 hours as this value is enforced as a maximum, but the bot role specifies a `MaxSessionTTL` of 12 hours, so this overrides the 24 hour value that is enforced.

Comment (strideynet) [reply lib/auth/join_test.go:409]: @timothyb89 - I can see a TODO in `createBotRole` - is this something we should reconsider now ?

**Review by ibeckermayer — APPROVED** [review]

**Review by jakule — APPROVED** [review]

**Comment by geekofalltrades** [comment]
I don't think this was ever backported to v9, even though it's labelled for it. I'm suffering this issue on v9.3.18.

**Comment by zmb3** [comment]
Hey @strideynet - can we get a v9 backport for this one?

**Comment by strideynet** [comment]
> Hey @strideynet - can we get a v9 backport for this one?

Yeah, will do this tomorrow morning.

**Comment by strideynet** [comment]
@geekofalltrades - that's been merged down into the V9 release branch now. It should be included in the next V9 release :) Apologies for the delay on this.

**Pull request merged**